tidy3d/components/tcad/boundary/heat.py

```diff
@@ -40,6 +53,137 @@ class HeatFluxBC(HeatChargeBC):
+        units=THERMAL_CONDUCTIVITY,
+    )
+    fluid_mu: pd.NonNegativeFloat = pd.Field(
+        title="Fluid Dynamic Viscosity",
+        description="Dynamic viscosity (μ) of the fluid.",
+        units=DYNAMIC_VISCOSITY,
+    )
+    fluid_Cp: pd.NonNegativeFloat = pd.Field(
+        title="Fluid Specific Heat",
+        description="Specific heat of the fluid at constant pressure.",
+        units=SPECIFIC_HEAT,
+    )
+    fluid_rho: pd.NonNegativeFloat = pd.Field(
+        title="Fluid Density",
+        description="Density (ρ) of the fluid.",
+        units=DENSITY,
+    )
+    fluid_beta: pd.NonNegativeFloat = pd.Field(
+        title="Fluid Thermal Expansivity",
+        description="Thermal expansion coefficient (β) of the fluid.",
+        units=THERMAL_EXPANSIVITY,
+    )
+    plate_L: pd.NonNegativeFloat = pd.Field(
+        title="Plate Characteristic Length",
+        description="Characteristic length (L), defined as the height of the vertical plate.",
+        units=MICROMETER,
+    )
+
+    gravity: pd.NonNegativeFloat = pd.Field(
+        default=9.80665 * 1e6,
+        title="Gravitational Acceleration",
+        description="Gravitational acceleration (g).",
+        units=ACCELERATION,
+    )
+
+    def from_si_units(
+        fluid_k: pd.NonNegativeFloat,
+        fluid_mu: pd.NonNegativeFloat,
+        fluid_Cp: pd.NonNegativeFloat,
+        fluid_rho: pd.NonNegativeFloat,
+        fluid_beta: pd.NonNegativeFloat,
+        plate_L: pd.NonNegativeFloat,
+        gravity: float = 9.80665,
+    ):
+        """
+        Create an instance from standard SI units.
+
+        Args:
+            fluid_k: Thermal conductivity in [W/(m*K)].
+            fluid_mu: Dynamic viscosity in [Pa*s].
+            fluid_Cp: Specific heat in [J/(kg*K)].
+            fluid_rho: Density in [kg/m**3].
+            fluid_beta: Thermal expansivity in [1/K].
+            plate_L: Plate characteristic length in [m].
+            gravity: Gravitational acceleration in [m/s**2].
+
+        Returns:
+            An instance of NaturalConvectionVerticalSpec with all values
+            converted to Tidy3D's internal unit system.
+        """
+
+        # --- Apply conversion factors ---
+        # value_tidy = value_si * factor
+        k_tidy = fluid_k / 1e6  # W/(m*K) -> W/(um*K)
+        mu_tidy = fluid_mu / 1e6  # Pa*s -> kg/(um*s)
+        cp_tidy = fluid_Cp * 1e12  # J/(kg*K) -> um**2/(s**2*K)
+        rho_tidy = fluid_rho / 1e18  # kg/m**3 -> kg/um**3
+        beta_tidy = fluid_beta  # 1/K -> 1/K (no change)
+        L_tidy = plate_L * 1e6  # m -> um
+        g_tidy = gravity * 1e6  # m/s**2 -> um/s**2
+
+        return NaturalConvectionVerticalSpec(
+            fluid_k=k_tidy,
+            fluid_mu=mu_tidy,
+            fluid_Cp=cp_tidy,
+            fluid_rho=rho_tidy,
+            fluid_beta=beta_tidy,
+            plate_L=L_tidy,
+            gravity=g_tidy,
+        )
+
+    def _compute_parameters(self):
```

Comment:
probably don't need it in the frontend (for example, if you happen to need to fix a bug in here, you would need to change both frontend and backend, instead of just backend)